models/articles.model.js

```diff
@@ -1,27 +1,45 @@
 const db = require("../db/connection.js");
-const {
-  validateArticleId,
-  validateIncVotes,
-} = require("../utils/utils.validation.js");
+const { validateIncVotes } = require("../utils/utils.validation.js");
 const { articleExists } = require("./utils.model.js");
 
-exports.fetchArticles = () => {
-  const queryStr = `
+exports.fetchArticles = (sort_by = "created_at", order = "desc") => {
+  const validSortColumns = [
+    "article_id",
+    "title",
+    "author",
+    "body",
+    "topic",
+    "created_at",
+    "votes",
+    "comment_count",
+  ];
+  const validOrder = ["asc", "desc", "ASC", "DESC"];
+
+  if (!validSortColumns.includes(sort_by) || !validOrder.includes(order)) {
+    return Promise.reject({
+      status: 400,
+      msg: "400 - Bad Request: Invalid sort_by or order query parameter",
+    });
+  }
+
+  let queryStr = `
   SELECT articles.article_id, articles.title, articles.topic, articles.author, articles.created_at, articles.votes, articles.article_img_url,
   COUNT(comments.comment_id) AS comment_count
   FROM articles
   LEFT JOIN comments ON comments.article_id = articles.article_id
+`;
+
+  queryStr += `
   GROUP BY articles.article_id
-  ORDER BY articles.created_at DESC;
+  ORDER BY ${sort_by} ${order.toUpperCase()};
 `;
-  return db
-    .query(queryStr)
-    .then(({ rows: articles }) => {
-      return articles;
-    })
-    .catch((err) => {
-      return Promise.reject(err);
-    });
+
+  return db.query(queryStr).then(({ rows }) => {
+    return rows.map((row) => ({
+      ...row,
+      comment_count: Number(row.comment_count),
```

Comment:
There is a way to ensure that this is a number when it is returned from the query! :) Have a look into the CAST sql command 😄

Reply:
Nice one Niamh! Thank you for pointing this out. :)